frontend/src/components/organisms/About.tsx

```diff
@@ -1,108 +1,177 @@
-    aliquip ex ea commodo consequat. Duis aute irure dolor in
-    reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
-    pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
-    culpa qui officia deserunt mollit anim id est laborum.
-  </p>
-  <h2>Programs</h2>
-  <h2>Certificate Request Process</h2>
-  <p>
-    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
-    eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim
-    ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
-    aliquip ex ea commodo consequat. Duis aute irure dolor in
-    reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
-    pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
-    culpa qui officia deserunt mollit anim id est laborum.
-  </p>
-  <h2>Request Certificate</h2>
-  <p>Fill out this form. Login first</p>
-  `;
-  const [value, setValue] = useState(`${default_text}`);
-  const [open, setOpen] = React.useState(false);
+
+  /** Tanstack query for fetching the about page data */
+  const {
+    data: aboutPageDetailsQuery,
+    isPending: aboutPageFetchPending,
+    isError,
+  } = useQuery({
+    queryKey: ["about"],
+    queryFn: async () => {
+      const { response, data } = await api.get("/about");
+      // console.log(data);
+      setValue(data.content);
+      return data;
+    },
+  });
+
+  let {
+    pageid,
+    content,
+  }: aboutPageData = {
+    pageid: aboutPageDetailsQuery?.id,
+    content: aboutPageDetailsQuery?.content,
+  };
+
+  const queryClient = useQueryClient();
+
+  /** Tanstack query mutation for changing the about page content */
+  const { mutateAsync: changeAboutContent } = useMutation({
+    mutationFn: async (variables: { newContent: string }) => {
+      const { newContent } = variables;
+      const { data } = await api.patch(`/${pageid}/about`, {
+        newContent: newContent,
+      });
+      return data;
+    },
+    retry: false,
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["about"] });
+    },
+    onError: (e: Error) => {
+      console.log(e.message);
+    },
+  });
+
+  const [value, setValue] = useState(content);
+  // console.log(`value: ${value}`);
+  const [openConfirm, setOpenConfirm] = React.useState(false);
   const [editMode, setEditMode] = React.useState(false);
-  const handleOpen = () => setOpen(true);
-  const handleClose = () => setOpen(false);
-  const handleClick = () => setEditMode(true);
+  const lastValueRef = useRef(value);
+  const handleConfirmOpen = () => setOpenConfirm(true);
+  const handleConfirmClose = () => setOpenConfirm(false);
+  const handleEditOpen = () => setEditMode(true);
+  const handleEditClose = () => {
+    setEditMode(false);
+    setValue(lastValueRef.current);
+  };
   const handleModalClick = () => {
     setEditMode(false);
-    handleClose();
+    handleConfirmClose();
+    lastValueRef.current = value;
```

Comment:
not sure if this is necessary, why do you need it?

Reply:
Because before this, when you edit you will edit the about content directly, so when you close it the changes will still be saved. I added this to ensure that when you don't want to save the edit and want to revert to the old edit, the old value is still there for us to revert back to.